p2p/transport/webrtc/internal/util.go

```diff
@@ -0,0 +1,56 @@
+package internal
+
+import (
+	"context"
+
+	"github.com/libp2p/go-libp2p/p2p/transport/webrtc/internal/encoding"
+	ma "github.com/multiformats/go-multiaddr"
+	"github.com/multiformats/go-multibase"
+	mh "github.com/multiformats/go-multihash"
+	"github.com/pion/datachannel"
+	"github.com/pion/webrtc/v3"
+)
+
+func DecodeRemoteFingerprint(maddr ma.Multiaddr) (*mh.DecodedMultihash, error) {
+	remoteFingerprintMultibase, err := maddr.ValueForProtocol(ma.P_CERTHASH)
+	if err != nil {
+		return nil, err
+	}
+	_, data, err := multibase.Decode(remoteFingerprintMultibase)
+	if err != nil {
+		return nil, err
+	}
+	return mh.Decode(data)
+}
+
+func EncodeDTLSFingerprint(fp webrtc.DTLSFingerprint) (string, error) {
+	digest, err := encoding.DecodeInterpersedHexFromASCIIString(fp.Value)
+	if err != nil {
+		return "", err
+	}
+	encoded, err := mh.Encode(digest, mh.SHA2_256)
+	if err != nil {
+		return "", err
+	}
+	return multibase.Encode(multibase.Base64url, encoded)
+}
+
+// only use this if the datachannels are detached, since the OnOpen callback
+// will be called immediately. Only use after the peerconnection is open.
+// The context should close if the peerconnection underlying the datachannel
+// is closed.
+func GetDetachedChannel(ctx context.Context, dc *webrtc.DataChannel) (rwc datachannel.ReadWriteCloser, err error) {
```

Comment:
Why is this in internal? Can't we move this to the webrtc package, as an unexported function?

Reply:
I could, but as it's used in two files and it's private I do not see the harm in it being in a child internal package. Is there some guideline against when to use internal or why you wouldn't do it? I mean in the end it will result in the same, no?

Reply:
In general, `internal` should be used sparingly, and only for things that logically should live in their own package. As such, helper function used by the `libp2pwebrtc` package should better be (unexported) functions in the same package, and not exported functions of an `internal` package.

Reply:
Got ya. Have moved this to its own file in the `libp2pwebrtc` package :)